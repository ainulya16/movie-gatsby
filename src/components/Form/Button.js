import React from 'react';
import { Button as ButtonDef, ButtonIcon } from '@aia-digital/ui-library';
import styled from 'styled-components';
import _ from 'lodash';
import { navigate } from '@reach/router';
import { Map, fromJS } from 'immutable';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { objectOf, any, string, instanceOf, func } from 'prop-types';
import API from '../../services/api';
import buildAPICALL from '../../helpers/buildCallAPI';
import { setOrderData } from '../../redux/Domain/action';
import { selectFormValues } from '../../redux/Form/selectors';
import selectDomain from '../../redux/Domain/selectors';

const CustomBodyList = {
  ORDERDATA: 'ORDERDATA',
};

const CustomCallbackList = {
  SETORDERDATA: 'SETORDERDATA',
};

const ButtonStyled = styled(ButtonDef)`
  width: ${({ theme }) => theme.color.w9};
  /* font: ${({ theme }) => theme.color.f9}; */
  font-size: 16px;
  border-radius: 4px;
  height: 50px;
  @media (max-width: 768px) {
    margin-bottom: 10px;
  }
`;
ButtonStyled.displayName = 'ButtonStyled';

const Button = ({
  label,
  customize,
  formdata,
  clicknext,
  clickprev,
  domainData,
  formReducerData,
  onSetOrderData,
}) => {
  const { action, componentProps, componentStyles } = customize;
  const domainValue = domainData.toJS();
  const formReducerDataValue = formReducerData.toJS();
  formdata = formdata ? formdata.toJS() : {};
  const source = {
    form: {
      ...formdata,
    },
  };

  const defaultActionFn = () => {};

  const oneMapCallBack = (response, target, callBack) => {
    const result = response.results[0] ? response.results[0] : {};
    if (result && !_.isEmpty(result)) {
      for (let index = 0; index < target.length; index++) {
        const element = target[index];
        callBack('application', element.field, result[element.source]);
      }
    }
  };

  const actionApiCall = () => {
    const callBackSource = [
      {
        service: 'https://developers.onemap.sg',
        fn: oneMapCallBack,
      },
    ];
    const callBack = action.fn;
    buildAPICALL(
      action.service,
      action.api,
      action.method || 'get',
      action.body || {},
      source || {},
    ).then(res => {
      const service = _.find(callBackSource, x => x.service === action.service);
      service.fn(res, action.customize.target, callBack);
    });
  };

  const bodyResolver = body => {
    const { ORDERDATA } = CustomBodyList;
    const orderData = {
      ...domainValue,
      application: _.get(formReducerDataValue, 'values', {}),
    };

    if (body.type === 'custom') {
      // for custom body
      switch (body.value) {
        case ORDERDATA:
          return orderData;

        default:
          return body.value;
      }
    }
    return body.value ? body.value : body;
  };

  const apiCall = async () => {
    const { path, url, method, body, endpoint } = action;
    const { SETORDERDATA } = CustomCallbackList;
    if (!path || !url || !method || !body || !endpoint) {
      console.log('Request is not valid');
    }
    const api = new API(endpoint);
    clicknext()
      .then(async () => {
        if (method === 'post') {
          const data = { path, method, url, body: bodyResolver(body) };
          try {
            const res = await api.post(data);
            const onSetOrderDataCB = _.find(
              action.customize.callback,
              cb => cb === SETORDERDATA,
            );
            // const onSetDomainDataCB = _.find(action.customize.callback, cb => cb === "onSetDomainData");
            if (!_.isEmpty(onSetOrderDataCB)) {
              onSetOrderData(res.data.orderReference);
            }
            return res;
          } catch (error) {
            console.log('Error', error);
            return error;
          }
        }
      })
      .catch(err => {
        console.log('Error', err);
        return err;
      });
  };

  const backFn = () => {
    const { url } = action;
    if (url) navigate(url);
    clickprev();
  };

  const sourceFn = [
    {
      name: 'populate',
      fn: actionApiCall,
    },
    {
      name: 'clicknext',
      fn: clicknext,
    },
    {
      name: 'clickprevious',
      fn: backFn,
    },
    {
      name: 'apiCall',
      fn: apiCall,
    },
  ];

  const actionFn = () => {
    const btnAction = !_.isEmpty(action)
      ? _.find(sourceFn, x => x.name === action.name)
      : { fn: defaultActionFn };
    btnAction.fn();
  };

  // if (customize.action && customize.action.type === 'api') {
  //   actionFn = actionApiCall;
  // } else {
  //   actionFn = defaultActionFn;
  // }

  // if (fieldValue.type && fieldValue.type === 'static') {
  //   radioValue = fieldValue.values;
  // }
  return customize.type && customize.type === 'icon' ? (
    <ButtonIcon onClick={actionFn} iconName={label} circle size="m" shadow />
  ) : (
    <ButtonStyled
      style={componentStyles}
      onClick={actionFn}
      {...componentProps}
    >
      {label}
    </ButtonStyled>
  );
};

Button.propTypes = {
  label: string,
  customize: objectOf(any).isRequired,
  domainData: instanceOf(Map),
  formReducerData: instanceOf(Map),
  formdata: instanceOf(Map),
  onSetOrderData: func,
  clicknext: func,
  clickprev: func,
};

Button.defaultProps = {
  label: 'Ok',
  domainData: fromJS({}),
  formReducerData: fromJS({}),
  formdata: fromJS({}),
  onSetOrderData: () => {},
  clicknext: () => {},
  clickprev: () => {},
};

export const mapStateToProps = createStructuredSelector({
  formReducerData: selectFormValues('application'),
  domainData: selectDomain(),
});

const mapDispatchToProps = {
  onSetOrderData: setOrderData,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Button);
