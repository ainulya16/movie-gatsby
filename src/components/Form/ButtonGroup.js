/* eslint-disable */
import React, { useEffect } from 'react';
import { Form } from 'antd';
import styled from 'styled-components';
import { ButtonGroup as ButtonGroupUI } from '@aia-digital/ui-library';
import withReduxField from './withReduxField';
import Label from '../Label';

const LabelButton = styled.div`
  font-family: 'AIARegular';
  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: 0.11px;
  color: #685556;
  padding-bottom: 15px;
`;

const Wrapper = styled.div`
  width: 100%;
  display: inherit;
  @media (max-width: 480px) {
    button {
      width: 100%;
    }
    button:nth-child(2) {
      margin-right: 0;
    }
  }
`;

const ButtonGroup = props => {
  const {
    label,
    labelCol,
    wrapperCol,
    extra,
    hasFeedback,
    colon,
    input: { name, onChange },
    meta: { invalid, error },
    fieldval,
    defaultValue,
  } = props;
  const hasError = invalid;

  const buttons = (fieldval.values || []).map(value => ({
    text: value.name,
    value: value.value,
  }));

  const onButtonChange = value => {
    onChange(value);
  };

  return (
    <Form.Item
      // label={<Label>{label}</Label>}
      wrapperCol={wrapperCol}
      labelCol={labelCol}
      help={hasError && <Label error>{error}</Label>}
      hasFeedback={hasFeedback && hasError}
      extra={extra}
      validateStatus={hasError ? 'error' : 'success'}
      colon={colon}
    >
      <LabelButton>{label}</LabelButton>
      <Wrapper>
        <ButtonGroupUI
          buttons={buttons}
          rounded
          radius="20px"
          space={15}
          buttonWidth={140}
          initialValue={
            defaultValue && defaultValue.value ? defaultValue.value : ''
          }
          onChange={onButtonChange}
          style={{ marginBottom: '20px' }}
        />
      </Wrapper>
    </Form.Item>
  );
};

export default withReduxField(ButtonGroup);
