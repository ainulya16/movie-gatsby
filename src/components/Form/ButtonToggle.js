import React from 'react';
import { Form } from 'antd';
import { ButtonToggle } from '@aia-digital/ui-library';
import {
  string,
  bool,
  number,
  func,
  objectOf,
  any,
  shape,
  arrayOf,
} from 'prop-types';
import styled, { css } from 'styled-components';
import Label from '../Label';
import withReduxField from './withReduxField';

const ButtonToggleContainer = styled.div`
  display: block;
  float: left;
  & div {
    ${props => props.inline && css`
      display: inline-block;
      margin-right: 20px;
    `}
  }
`;

ButtonToggleContainer.displayName = 'ButtonToggleContainer';

/**
 * 
 * ?USAGE
 * <CheckBoxGroup name="checkbox"
 * optionsList={[{text: 'email', value: 'email'}, {text: 'phone', value: 'phone'}]} 
 * onChange={e=>console.log(e)}/>
 */
const Input = ({
  label,
  labelCol,
  wrapperCol,
  help,
  extra,
  hasFeedback,
  colon,
  required,
  input,
  meta: { invalid, error },
  optionsList,
  inline,
  rounded,
  radius,
  theme,
  fieldval,
  customize,
  // buttonWidth,
  containerWidth,
  ...props
}) => {
  // const { buttonToggleWidth } = customize;
  const { componentClass } = customize;
  const hasError = invalid;
  const { onChange } = input;
  let buttons = [];
  if (fieldval && fieldval.type === 'static' && fieldval.values)
  buttons = fieldval.values.map(item => ({ text: item.name, value: item.value }));
  return (
    <Form.Item
      label={label && <Label> {label}</Label>}
      wrapperCol={wrapperCol}
      labelCol={labelCol}
      help={hasError && <Label error> {error}</Label>}
      hasFeedback={hasFeedback && hasError}
      extra={extra}
      validateStatus={hasError ? 'error' : 'success'}
      colon={colon}
      {...props}
      className={componentClass}
    >
      <ButtonToggle
        inline={inline}
        onChange={e => onChange(e)}
        buttons={buttons}
        rounded={rounded}
        radius={radius}
        containerWidth={containerWidth}
        // buttonWidth={buttonToggleWidth || buttonWidth}
        theme={theme}
      />
    </Form.Item>
  );
};

Input.propTypes = {
  label: string,
  labelCol: objectOf(number),
  wrapperCol: objectOf(number),
  help: string,
  extra: string,
  hasFeedback: bool,
  colon: bool,
  required: bool,
  input: shape({
    name: string,
    onBlur: func,
    onChange: func,
    onDragStart: func,
    onDrop: func,
    onFocus: func,
    value: string,
  }).isRequired,
  meta: shape({
    touched: bool,
    invalid: bool,
    error: string,
  }).isRequired,
  optionsList: arrayOf(shape({text: string, value: string})).isRequired,
  inline: bool,
  buttons: arrayOf(shape({
      text: string,
      value: string
  })).isRequired,
  rounded: bool,
  // buttonWidth: number,
  containerWidth: string,
  radius: string,
  theme: string,
  fieldval: objectOf(any),
  customize: objectOf(any),
};
Input.defaultProps = {
  label: null,
  labelCol: null,
  wrapperCol: null,
  help: '',
  extra: '',
  hasFeedback: false,
  colon: false,
  required: false,
  inline: true,
  rounded: true,
  // buttonWidth: 100,
  containerWidth: null,
  radius: '20px',
  theme: 'primary1',
  fieldval: {},
  customize: {},
};

export const InputTest = Input;
export default withReduxField(Input);