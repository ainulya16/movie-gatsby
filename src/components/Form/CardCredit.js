/* eslint-disable */
import React, { useEffect, useState } from 'react';
import { Form } from 'antd';
import _ from 'lodash';
import styled from 'styled-components';
import { CardCredit as CardCreditUI } from '@aia-digital/ui-library';
import withReduxField from './withReduxField';

const StyledCardCredit = styled(CardCreditUI)`
  @media (max-width: 375px) {
    width: 296px;
    height: 296px;
    padding: 16px 20px 20px 20px;
    background: red;
  }
`;

const CardCredit = ({
  label,
  labelCol,
  wrapperCol,
  extra,
  hasFeedback,
  colon,
  input,
  meta: { invalid, error },
  fieldval,
  defaultValue,
  ...attr
}) => {
  const { onChange, value } = input;

  const changeCardCreditValue = (val, name) => {
    onChange({ ...value, [name]: _.get(val, 'target.value', val) });
  };

  return (
    <StyledCardCredit
      {...attr}
      cardNumber={value.cardNumber}
      expiry={value.cardExpiry}
      name={value.cardName}
      cvv={value.cardCvv}
      changeCard={e => changeCardCreditValue(e, "cardNumber")}
      changeExpiry={e => changeCardCreditValue(e, "cardExpiry")}
      changeName={e => changeCardCreditValue(e, "cardName")}
      changeCvv={e => changeCardCreditValue(e, "cardCvv")}
    />
  );
};

export default withReduxField(CardCredit);
