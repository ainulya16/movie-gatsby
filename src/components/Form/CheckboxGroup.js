import React from 'react';
import { Form } from 'antd';
import { Checkbox } from '@aia-digital/ui-library';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import Label from '../Label';
import withReduxField from './withReduxField';

const CheckboxContainer = styled.div`
  display: block;
  float: left;
  & div {
    ${props => props.inline && css`
      display: inline-block;
      margin-right: 20px;
    `}
  }
`;

CheckboxContainer.displayName = 'CheckboxContainer';
const StyledCheckbox = styled(Checkbox)`
`;

/**
 * 
 * ?USAGE
 * <CheckBoxGroup name="checkbox"
 * optionsList={[{label: 'email', value: 'email'}, {label: 'phone', value: 'phone'}]} 
 * onChange={e=>console.log(e)}/>
 */
const Input = ({
  label,
  labelCol,
  wrapperCol,
  help,
  extra,
  hasFeedback,
  colon,
  required,
  input,
  meta: { invalid, error },
  optionsList,
  inline,
  ...props
}) => {
  const hasError = invalid;
  const { onChange } = input;
  return (
    <Form.Item
      label={label && <Label> {label}</Label>}
      wrapperCol={wrapperCol}
      labelCol={labelCol}
      help={hasError && <Label error> {error}</Label>}
      hasFeedback={hasFeedback && hasError}
      extra={extra}
      validateStatus={hasError ? 'error' : 'success'}
      colon={colon}
      {...props}
    >
      <CheckboxContainer inline={inline}>
      {optionsList.map((item, index) => {
        const isChecked = input.value.indexOf(item.value) !== -1;
        return (
          <StyledCheckbox
            key={item.value}
            name={`${input.value}-${index}`}
            value={item.value}
            checked={isChecked}
            onChange={(e)=>{
              const { checked } = e.target;
              const value = [...input.value];
              if(checked){
                value.push(item.value);
              } else {
                value.splice(value.indexOf(item.value), 1);
              }
              onChange(value);
            }}
          >{item.label}
          </StyledCheckbox>
        );
      })}
      </CheckboxContainer>
    </Form.Item>
  );
};

Input.propTypes = {
  label: PropTypes.string,
  labelCol: PropTypes.objectOf(PropTypes.number),
  wrapperCol: PropTypes.objectOf(PropTypes.number),
  help: PropTypes.string,
  extra: PropTypes.string,
  hasFeedback: PropTypes.bool,
  colon: PropTypes.bool,
  required: PropTypes.bool,
  input: PropTypes.shape({
    name: PropTypes.string,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onDragStart: PropTypes.func,
    onDrop: PropTypes.func,
    onFocus: PropTypes.func,
    value: PropTypes.string,
  }).isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    invalid: PropTypes.bool,
    error: PropTypes.string,
  }).isRequired,
  optionsList: PropTypes.arrayOf(PropTypes.shape({label: PropTypes.string, value: PropTypes.string})).isRequired,
  inline: PropTypes.bool
};
Input.defaultProps = {
  label: null,
  labelCol: null,
  wrapperCol: null,
  help: '',
  extra: '',
  hasFeedback: false,
  colon: false,
  required: false,
  inline: true,
};

export const InputTest = Input;
export default withReduxField(Input);