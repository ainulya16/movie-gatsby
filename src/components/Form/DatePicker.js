import React from 'react';
import { Form } from 'antd';
import { DatePicker as DatePickerBase } from '@aia-digital/ui-library';
import {
  objectOf,
  string,
  any,
  number,
  bool,
  shape,
  func,
  object,
} from 'prop-types';
import moment from 'moment';
import styled from 'styled-components';
import Label from '../Label';
import withReduxField from './withReduxField';
import { setDateConfigByAge } from '../../helpers/helper_functions';

export const StyledFormItem = styled(Form.Item)`
  .ant-form-item-control-wrapper .ant-form-item-control.has-error span input {
    border: ${({ theme }) => `
      1px solid ${theme.color.u5} !important
    `};
  }

  & div {
    span {
      width: ${props => props.inputWidth || 'auto'};
    }
  }

  .ant-calendar-picker-input {
    width: 100%;
    height: ${({ theme }) => theme.color.h9};
    font: ${({ theme }) => theme.color.f9};
    color: ${({ theme }) => theme.color.c9};
    background-color: ${({ theme }) => theme.color.b9};
  }
`;

const disabledDateByConfig = (
  current,
  { minAge, maxAge, dateFormat = 'DD/MM/YYYY' },
) => {
  if (!minAge || !maxAge) {
    return false;
  }

  const minDateCondition =
    current > moment(setDateConfigByAge(minAge), dateFormat);
  const maxDateCondition =
    current < moment(setDateConfigByAge(maxAge), dateFormat);
  return maxDateCondition || minDateCondition;
};

export const StyledInput = styled(DatePickerBase)`
  && .ant-input {
    padding: 8px 16px;
    border-radius: ${({ theme }) => theme.color.br9};
    width: ${props => props.inputWidth || '335px'};
    height: ${({ theme }) => theme.color.h9};
    font: ${({ theme }) => theme.color.f9};
    line-height: 17px !important;
    color: ${({ theme }) => theme.color.c9};
    background-color: ${({ theme }) => theme.color.b9};
    font-weight: 400;
    border: ${({ theme }) => theme.color.bd9 || '1 px solid #ccc'};
    &:focus {
      color: ${({ theme }) => theme.color.c9};
      border: ${({ theme }) => theme.color.bd9 || '1 px solid #ccc'};
      outline: none;
      -webkit-box-shadow: none;
      box-shadow: none;
    }
  }
`;

const DatePicker = ({
  label,
  labelCol,
  wrapperCol,
  help,
  extra,
  hasFeedback,
  colon,
  required,
  input,
  meta: { invalid, error },
  dateFormat,
  minAge,
  maxAge,
  customize,
  ...props
}) => {
  const { componentStyles, inputWidth, componentClass } = customize;
  const hasError = invalid;
  const onChange = (date, dateString) => {
    input.onBlur({
      target: {
        value: dateString,
      },
    });
    return input.onChange(dateString);
  };
  const defaultValue = moment(setDateConfigByAge(minAge), dateFormat);
  return (
    <StyledFormItem
      label={label && <Label>{label}</Label>}
      wrapperCol={wrapperCol}
      labelCol={labelCol}
      help={hasError && <Label error>{error}</Label>}
      hasFeedback={hasFeedback && hasError}
      extra={extra}
      validateStatus={hasError ? 'error' : 'success'}
      colon={colon}
      inputWidth={inputWidth}
      className={componentClass}
    >
      <StyledInput
        style={componentStyles}
        inputWidth={inputWidth}
        // disabledDate={current =>
        //   disabledDateByConfig(current, { minAge, maxAge, dateFormat })
        // }
        defaultValue={defaultValue}
        onChange={onChange}
        format={dateFormat}
        size="large"
      />
    </StyledFormItem>
  );
};

DatePicker.propTypes = {
  config: objectOf(any),
  label: string,
  labelCol: objectOf(number),
  wrapperCol: objectOf(number),
  help: string,
  extra: string,
  hasFeedback: bool,
  colon: bool,
  required: bool,
  input: shape({
    name: string,
    onBlur: func,
    onChange: func,
    onDragStart: func,
    onDrop: func,
    onFocus: func,
    value: object,
  }).isRequired,
  meta: shape({
    touched: bool,
    invalid: bool,
    error: string,
  }).isRequired,
  customize: objectOf(any),
};

DatePicker.defaultProps = {
  label: null,
  labelCol: null,
  wrapperCol: null,
  help: '',
  extra: '',
  hasFeedback: false,
  colon: false,
  required: false,
  config: {},
  customize: {},
};

export const InputTest = DatePicker;
export default withReduxField(DatePicker);
