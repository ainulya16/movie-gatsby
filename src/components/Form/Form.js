import React from 'react';
import { compose, withProps } from 'recompose';
import { reduxForm } from 'redux-form/immutable';
import PropTypes from 'prop-types';
import { Form as BaseForm } from 'antd';

export const Form = ({ children, handleSubmit }) => (
  <BaseForm onSubmit={handleSubmit}> {children}</BaseForm>
);

Form.propTypes = {
  children: PropTypes.node.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default compose(
  withProps(),
  reduxForm(),
)(Form);
