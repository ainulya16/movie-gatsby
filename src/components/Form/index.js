// import { ButtonGroup } from '@aia-digital/ui-library';
import Form from './Form'; // eslint-disable-line
import Input from './Input';
import Select from './Select';
import Radio from './Radio';
import Button from './Button';
import CheckBoxGroup from './CheckboxGroup';
import DatePicker from './DatePicker';
import PhoneNumber from './PhoneNumber';
import ButtonToggle from './ButtonToggle';
import ButtonGroup from './ButtonGroup';
import InputPhoneNumber from './InputPhoneNumber';
import Search from './Search';
import CardCredit from './CardCredit';
import Title from './Title';

export {
  CardCredit,
  Input,
  Select,
  DatePicker,
  Radio,
  Button,
  PhoneNumber,
  CheckBoxGroup,
  ButtonGroup,
  ButtonToggle,
  InputPhoneNumber,
  Search,
  Title,
};

export default Form;
