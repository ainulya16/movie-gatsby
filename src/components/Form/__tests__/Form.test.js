import React from 'react';
// import jest-styled-components for snapshot styled components
import 'jest-styled-components';
import renderer from 'react-test-renderer';

// import components to be tested
import { Form } from '../Form';

describe('Input Component', () => {
  it('Form component should match the snapshot', () => {
    const tree = renderer
      .create(
        <Form handleSubmit={jest.fn()}>
          {' '}
          <h1>test</h1>{' '}
        </Form>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
