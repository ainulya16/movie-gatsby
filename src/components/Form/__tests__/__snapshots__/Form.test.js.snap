// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Form component should match the snapshot 1`] = `
<form
  className="ant-form ant-form-horizontal"
  onSubmit={[MockFunction]}
>
   
   
  <h1>
    test
  </h1>
   
</form>
`;
